File: cld/src/main/java/cloudcmd/cld/commands/Verify.java
package cloudcmd.cld.commands;


import cloudcmd.common.JsonUtil;
import cloudcmd.common.config.ConfigStorageService;
import cloudcmd.common.engine.CloudEngineService;
import cloudcmd.common.index.IndexStorageService;
import jpbetz.cli.Command;
import jpbetz.cli.CommandContext;
import jpbetz.cli.Opt;
import jpbetz.cli.SubCommand;
import org.json.JSONArray;
import org.json.JSONObject;

@SubCommand(name = "verify", description = "Verify storage contents.")
public class Verify implements Command
{
  @Opt(opt = "n", longOpt = "minTier", description = "min tier to verify to", required = false)
  Number _minTier = 0;

  @Opt(opt = "m", longOpt = "maxTier", description = "max tier to verify to", required = false)
  Number _maxTier = Integer.MAX_VALUE;

  @Opt(opt = "a", longOpt = "all", description = "verify all", required = false)
  boolean _verifyAll = true;

  @Opt(opt = "d", longOpt = "delete", description = "delete invalid blocks", required = false)
  boolean _deleteOnInvalid = true;

  @Override
  public void exec(CommandContext commandLine) throws Exception
  {
    JSONArray selections = _verifyAll ? IndexStorageService.instance().find(new JSONObject()) : JsonUtil.loadJsonArray(System.in);
    CloudEngineService.instance().verify(_minTier.intValue(), _maxTier.intValue(), selections, _deleteOnInvalid);
  }
}

File: cld/src/main/scala/cloudcmd/cld/commands/Verify.scala
package cloudcmd.cld.commands

import cloudcmd.cld.CloudServices
import cloudcmd.common.util.JsonUtil
import jpbetz.cli.Command
import jpbetz.cli.CommandContext
import jpbetz.cli.Opt
import jpbetz.cli.SubCommand
import org.json.JSONObject

@SubCommand(name = "verify", description = "Verify storage contents.") class Verify extends Command {

  @Opt(opt = "n", longOpt = "minTier", description = "min tier to verify to", required = false) private var _minTier: Number = 0
  @Opt(opt = "m", longOpt = "maxTier", description = "max tier to verify to", required = false) private var _maxTier: Number = Integer.MAX_VALUE
  @Opt(opt = "a", longOpt = "all", description = "verify all", required = false) private var _verifyAll: Boolean = true
  @Opt(opt = "d", longOpt = "delete", description = "delete invalid blocks", required = false) private var _deleteOnInvalid: Boolean = true

  def exec(commandLine: CommandContext) {
    val selections = if (_verifyAll) CloudServices.IndexStorage.find(new JSONObject) else JsonUtil.loadJsonArray(System.in)
    CloudServices.CloudEngine.filterAdapters(_minTier.intValue, _maxTier.intValue)
    CloudServices.IndexStorage.verify(selections, _deleteOnInvalid)
  }
}